### PR TITLE
Make Buildifier Sanity Test Strict

### PR DESCRIPTION
**Pull request by gnossen**
This should help avoid rollbacks due to internal buildifier discrepancies in the future.

This PR also manually modifies Starlark and BUILD files to make them conform to the stricter rules.

### CODE REVIEW

**Comment by gnossen** [comment]
Failures:
 - [Artifact Build MacOS](https://github.com/grpc/grpc/issues/27932)

**Comment by gnossen** [comment]
Import flake:
- //third_party/py/grpc:tests_aio/unit/server_test_opensource (totally unrelated)

File: tools/distrib/buildifier_format_code.sh [hunk 1]
```diff
@@ -15,9 +15,9 @@
 
 set -e
 
-BUILDIFIER_VERSION="0.29.0"
+BUILDIFIER_VERSION="4.2.2"
```
Comment (lidizheng) [inline tools/distrib/buildifier_format_code.sh:18]: Wow, they bumped 4 major versions in 2 years.

File: tools/distrib/buildifier_format_code_strict.sh [hunk 2]
```diff
@@ -17,4 +17,4 @@
 dir=$(dirname "${0}")
 buildifier_format_script="${dir}/buildifier_format_code.sh"
 
-${buildifier_format_script} -lint warn
+${buildifier_format_script} -lint fix
```
Comment (lidizheng) [inline tools/distrib/buildifier_format_code_strict.sh:20]: Should we just add the `-lint fix` to the `buildifier_format_code.sh` script, and remove this file?

Comment (gnossen) [reply tools/distrib/buildifier_format_code_strict.sh:20]: I've found it useful to have a script that proxies flags to buildifier verbatim. The problem with adding `-lint fix` unconditionally is that there are certain other flags that cannot be used with it.

**Pull request merged**